src/phoenix/server/app.py

```diff
@@ -356,29 +380,33 @@ def create_app(
     tracer_provider = None
     strawberry_extensions = schema.get_extensions()
     if server_instrumentation_is_enabled():
-        from opentelemetry.instrumentation.sqlalchemy import SQLAlchemyInstrumentor
-        from opentelemetry.trace import TracerProvider
-        from strawberry.extensions.tracing import OpenTelemetryExtension
+        if engine := db.engine:
+            from opentelemetry.instrumentation.sqlalchemy import SQLAlchemyInstrumentor
+            from opentelemetry.trace import TracerProvider
+            from strawberry.extensions.tracing import OpenTelemetryExtension
+
+            tracer_provider = initialize_opentelemetry_tracer_provider()
+            SQLAlchemyInstrumentor().instrument(
+                engine=engine.sync_engine,
+                tracer_provider=tracer_provider,
+            )
+            clean_ups.append(SQLAlchemyInstrumentor().uninstrument)
+            if TYPE_CHECKING:
+                # Type-check the class before monkey-patching its private attribute.
+                assert OpenTelemetryExtension._tracer
```

Comment:
How come we have this inside a `TYPE_CHECKING` block? This means this assertion statement will run when `mypy` runs?

Reply:
I'm not sure, this is relocated code

Reply:
`TYPE_CHECKING` evaluates to true when a type check is running. Not sure what this is intended to do. Maybe a mistake?

Reply:
My guess is that it's a hint for the type checker

Reply:
This is to check that the private `_tracer` class [variable](https://github.com/strawberry-graphql/strawberry/blob/fc322a8ab921e76c9f790633693a9e2b8906a711/strawberry/extensions/tracing/opentelemetry.py#L41) still exist. The runtime set operation, i.e. `self._tracer = ...`, below will succeed regardless.